src/bridge/cockpitdbusjson.c

```diff
@@ -2917,24 +2956,48 @@ cockpit_dbus_json_prepare (CockpitChannel *channel)
       else
         self->logname = bus;
 
-      /* Ready when the bus connection is available */
-      if (self->bus_type == G_BUS_TYPE_NONE)
+      if (address == NULL && self->bus_type != G_BUS_TYPE_NONE)
         {
-          GDBusConnectionFlags flags = G_DBUS_CONNECTION_FLAGS_AUTHENTICATION_CLIENT;
-          if (self->default_name)
-            flags = flags | G_DBUS_CONNECTION_FLAGS_MESSAGE_BUS_CONNECTION;
+          bus_address = g_dbus_address_get_for_bus_sync (self->bus_type, self->cancellable, NULL);
+          address = bus_address;
+        }
 
```

Comment:
When G_BUS_TYPE_NONE, we should just open the connection as before, and not use shared connections. These are typically peer to peer connections, and should not start being shared.

```

-          g_dbus_connection_new_for_address (address,
-                                             flags,
-                                             NULL,
-                                             self->cancellable,
-                                             on_connection_ready,
-                                             g_object_ref (self)); 
```